[#2] 회원가입 기능 구현

ksjeong761 opened the pull request.

ksjeong761 committed:
[#2] 데이터 접근 계층에 Repository 대신 Mapper를 바로 사용

ksjeong761 committed:
[#2] 리뷰 반영 : Controller 관련
- 컨트롤러에 @RequestMapping 따로 안붙여도 됨
- 생성자 주입에는 @Autowired 따로 안붙여도 됨
- Controller 단위 테스트에는 MockMvc의 standalone setup 사용 제안
- @ModelAttribute보다는 @RequestBody 사용 제안

ksjeong761 committed:
[#1] 설정 파일 확장자 변경 properties -> yaml

ksjeong761 committed:
[#2] 회원가입 검증 - Controller

ksjeong761 committed:
[#2] 회원가입 검증 - Mapper

Joel-FLab reviewed (changes requested):
몇 가지 질문과 몇 가지 변경 제안드렸습니다 ㅎㅎ

ksjeong761 committed:
[#2] 리뷰 반영
- 회원가입 성공 시 ResponseEntity 말고 객체로 반환
- User 클래스를 UserRequestDTO, UserResponseDTO로 변경
- 불필요한 로그 삭제

- 테스트 클래스에서 스프링 Bean 제거
- 한 번에 하나씩만 테스트하도록 단위 테스트 분리
- 테스트 네이밍 컨벤션 적용
